### ISSUE TITLE
Final Fantasy IX (377840)

### ISSUE THREAD

**Issue by adrianlshaw** [ValveSoftware/Proton#708]
Using Proton 3.7-3.

After the "preparing to launch" Steam popup, the game silently exits without any error or additional popup. No launcher or game.

Ubuntu 18.04, NVIDIA Quadro 1800
Full specs: https://gist.github.com/adrianlshaw/13b21eae4d9ad3c6c79474396110feed

**Comment by Zero86Sk** [comment]
Just tried it myself. I just had a crash to the desktop. No Launcher or Game.
Proton 3.7.4 Beta

my steam info:
https://gist.github.com/Zero86Sk/d629f657c1ae861c36657cdf238a7765

**Comment by DistantThunder** [comment]
The game launcher crashes indeed.

Using the latest Beta.


System info attached.
Report attached.

[steam_system_info.txt](https://github.com/ValveSoftware/Proton/files/2341656/steam_system_info.txt)
[steam-377840.log](https://github.com/ValveSoftware/Proton/files/2341649/steam-377840.log)

**Comment by jmf1sh** [comment]
I get the same behavior as well. However running it directly with wine-staging using a 32-bit prefix works perfectly.

**Comment by DistantThunder** [comment]
Referencing #18

**Comment by artumino** [comment]
I can confirm that as stated at some point in issue #18, installing .NET and wine-mono does indeed let the game run.

If you have all the necessary packages (wine-staging, wine-mono, etc.)
You can follow [this guide](https://www.reddit.com/r/wine_gaming/comments/8r6low/guide_how_to_install_net_45_on_64bit_prefixes/) on how to enable it in a generic wine prefix.

The Wine prefix for FFIX is under "steamapps/compatdata/377840/pfx".
After installing .NET Framework 4 and 4.7 I had to run the .NET Framework 4.5.2 installer under "steamapps/common/FINAL\ FANTASY\ IX/_CommonRedist/DotNet/4.5.2/" in order for the game to load.
I can now launch the game from Steam just fine and everything seems to work...

**Comment by kisak-valve** [comment]
Final Fantasy IX - No Sound

Issue transferred from https://github.com/ValveSoftware/Proton/issues/2349.
@Pantar2205 posted on 2019-02-19T16:32:30:

# Compatibility Report
- Name of the game with compatibility issues: Final Fantasy IX
- Steam AppID of the game: 377840

## System Information
- GPU: RX480
- Driver/LLVM version: Mesa18.3.3
- Kernel version: 4.20.10-arch1-1-ARCH
- Proton version: 3.16-4

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.



## Symptoms 
No Sound

Tried installing directsound via winetricks [This fixed sound in FF8] but no luck. Can't find anything else on protondb or in this github. Steam Controller and graphics are flawless, even on 4k monitor.

**Comment by Pantar2205** [comment]
Follow up.

two machines =

machine A
e3-1231v3 // rx550 // arch //gnome // = game is perfect, sound is perfect, steam controller stuck pointing up [only this game]

machine B
ryzen 1700 // rx480 // arch // kde = game is perfect, controller is perfect, no sound.

will keep digging for now.

**Comment by shoober420** [comment]
# Compatibility Report
- Name of the game with compatibility issues: FINAL FANTASY IX
- Steam AppID of the game: https://steamdb.info/app/377840/

## System Information
- GPU: GTX 760
- Driver/LLVM version: NVIDIA 440.64
- Kernel version: 5.5.8-zen
- https://github.com/shoober420/linux-scripts/blob/main/home/shoober420/sysinfonvidia
- Proton version: 5.0-3

LOG:
[steam-377840.log](https://github.com/ValveSoftware/Proton/files/4301489/steam-377840.log)


## Symptoms
Crashes at launch

**Comment by Eoin-ONeill-Yokai** [comment]
> Final Fantasy IX - No Sound
> 
> Issue transferred from #2349.
> @Pantar2205 posted on 2019-02-19T16:32:30:
> # Compatibility Report
> 
>     * Name of the game with compatibility issues: Final Fantasy IX
> 
>     * Steam AppID of the game: 377840
> 
> 
> ## System Information
> 
>     * GPU: RX480
> 
>     * Driver/LLVM version: Mesa18.3.3
> 
>     * Kernel version: 4.20.10-arch1-1-ARCH
> 
>     * Proton version: 3.16-4
> 
> 
> ## I confirm:
> 
>     * [x]  that I haven't found an existing compatibility report for this game.
> 
>     * [x]  that I have checked whether there are updates for my system available.
> 
> 
> ## Symptoms
> 
> No Sound
> 
> Tried installing directsound via winetricks [This fixed sound in FF8] but no luck. Can't find anything else on protondb or in this github. Steam Controller and graphics are flawless, even on 4k monitor.

@kisak-valve On KDE there are two entries for Final Fantasy IX on the `Sound -> Applications` menu. It seems like KDE is confused and believes there are actually two different Final Fantasy IX processes generating sound. 

Should be worth noting that protonfixes with this game works pretty well other than the sound issue on KDE, which might just be a quirk of the DE. What is valve's policy regarding including protonfixes as a community driven patching tool with the default proton distribution?

**Comment by ugly95** [comment]
The game starts normally with 5.13.  The game appears to be playable now.

**Comment by adrianlshaw** [comment]
Just came here to say that Proton 6.3-1 fixes all the launcher issues - one of the long remaining issues with FF9. Well done Valve :+1:  now the game runs well without modification

**Comment by Wedge009** [comment]
Only started playing this on 20th July 2021, with Proton 6.3-5, so everything's been flawless for me so far - no launcher issues or anything like that. Am about a third of the way through the story.

Oh, I can confirm @Eoin-ONeill-Yokai's observation regarding the double entries for the FFIX application in KDE audio, but only the second one is effective. Not sure if that's specific to Proton or the game itself - I've seen some games, even native ones, have duplicate entries or list a second entry for a platform like FMOD, etc. The sounds itself works fine, though, at least on Proton 6.3.

**Comment by Eoin-ONeill-Yokai** [comment]
Having started the game again recently on a fresh install, I've noticed a regression on Proton 6.3-5 where the game isn't saving correctly. 

I've played again up to the first save point (Chapel in Alexandria) and tried to save the game and quit. Next time I opened the game up, all the save slots are empty. I was able to reproduce it again to confirm it.

**Comment by kisak-valve** [comment]
Hello @Eoin-ONeill-Yokai, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options, reproduce the regression, and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.)

**Comment by Eoin-ONeill-Yokai** [comment]
Hi @kisak-valve ,

I've done more experimenting and saved a log file. (Tangential note, is there a way to specify a desired Log folder for proton?)

I tried a few more things this time just to see if I could tell what's going on. It seems to me like the save functionality is off-loaded to another part of the game as saving the game and returning to the main menu will allow you to load the game as usual. However, after closing the game (and subsequently relaunching), the game will lose all save data.

Here's a summary of the proton log (gzipped):
[1st-run-steam-377840.log.gz](https://github.com/ValveSoftware/Proton/files/6951597/1st-run-steam-377840.log.gz)

And, since I'm not sure entirely if the issue occurs on launch or on close, I've also included a second run. This second run is when loading the game, seeing the menu, and witnessing no save files available.
[2nd-run-steam-377840.log.gz](https://github.com/ValveSoftware/Proton/files/6951601/2nd-run-steam-377840.log.gz)

Another important note for testing is that the game's built in "checkpointing" system (represented as "Continue" on the main menu) also resets. This means that you could probably test this quickly by simply playing far enough to trigger the first checkpoint and then closing and reopening the game. I can't recall where this happens, but I think getting to Vivi after the Zidane controlled introduction might be a safe place to start. This is useful, since the first save moogle requires going through a few events that can take a few minutes.

My intuition would be that the game is trying to move a set of files form `$compatuser/Temp/SquareEnix/Final Fantasy IX/ ` to `$compatuser/AppData/LocalLow/SquareEnix/FINAL FANTASY IX/` when it terminates, but it's failing silently and losing all progress as a result. I've noticed that an `AppData` folder doesn't exist, and instead is represented by `Application Data` -- perhaps this could be the source of the problem? I could always try making the desired save directories to see if that helps straighten out the issue.

Anyway, hope that wasn't too long and thanks for taking the time to read this. I know it's busy over there at Valve right now. Cheers!

**Comment by Wedge009** [comment]
I believe auto-save or 'check-pointing' occurs with every screen load, ie moving to a new area in town, or entering/exiting the field in the main map.

But I've been playing entirely on Proton 6.3-5 and don't observe (at least not yet) any issue with saving or loading, so I'm a bit confused as to what's going on here...

**Comment by Eoin-ONeill-Yokai** [comment]
@Wedge009 Interesting. I guess it should be worth noting I'm using a BTRFS system, though I really doubt that would affect it. It is something that has changed since the last time I played this game though...

**Comment by alasky17** [comment]
@Eoin-ONeill-Yokai Were you using a BTRFS system when you first played and the saves were working?  Do you have anywhere to test with a more standard file system on the same hardware?

**Comment by madewokherd** [comment]
> I've done more experimenting and saved a log file. (Tangential note, is there a way to specify a desired Log folder for proton?)

Yes. https://github.com/ValveSoftware/Proton/issues/4265#issuecomment-722626842

**Comment by Eoin-ONeill-Yokai** [comment]
Let me start by saying I fixed this issue by symlinking `AppData` to  `Application Data` and making a `LocalLow` folder inside. Therefore, if anybody has that issue in the future, just ensure you have a directory `.../pfx/drive_c/users/steamuser/AppData/LocalLow` present and the save files will work as expected.


> @Eoin-ONeill-Yokai Were you using a BTRFS system when you first played and the saves were working? Do you have anywhere to test with a more standard file system on the same hardware?

@alasky17 No I wasn't. I took some time to test it on another machine in my house running Fedora (on ext4) and can confirm that this isn't occurring on that machine. So the problem seems to stem from the use of BTRFS on my Manjaro machine.

I guess I'm mostly confused by why the process would be different -- perhaps it could be that something about the CoW process is messing with Proton's ability to setup the directories in `steamuser` folder properly -- either that or I should run more diagnostics on my new NVME drive to ensure that there's nothing faulty going on. ;) 

Anyway, that **probably** means that this being a regression is a false alarm. Sorry about that, I think the BTRFS element of this new install was working so well that I kind of forgot to mention it. Let me just say that the process of launching this game is much better than in prior proton versions, and even installing mods is relatively easy compared to prior attempts (though moguri's installer is still broken, for those who are curious.)


> > I've done more experimenting and saved a log file. (Tangential note, is there a way to specify a desired Log folder for proton?)
> 
> Yes. [#4265 (comment)](https://github.com/ValveSoftware/Proton/issues/4265#issuecomment-722626842)

@madewokherd Nice! I'll probably set that globally to my temp folder then since I often don't need them to persist, but I like the option of checking after launch. :)

**Comment by Wedge009** [comment]
Didn't notice any issues after a quick test with Proton 7.0-1.

**Comment by sekoku** [comment]
> Didn't notice any issues after a quick test with Proton 7.0-1.

Not to bump an old thread (apologizes) but the issue isn't resolved with 7.0-3 (available via Steam Proton choices). It crashes/doesn't launch the launcher or game. Even the newest proton versions (9, 10, 11) are similar "out of box" and it's a known thing to where "ProtonDB" recommends using a Mod to get around this.

Unfortunately, unless Valve or the Proton/DXVK devs talk to Square(-Enix) and/or do the work of making the initial "install" script/launcher past initial boot/install work "out of box" before having to worry about modding the game for "improvement" purposes, the game is **unplayable on Linux** as is right now. :( (Unless you mod, but I'm going by OOB experience)

I'm attaching the log for Proton-11 (removing my user information) in case someone at Valve is able to resolve this.

[steam-377840.log](https://github.com/user-attachments/files/28975695/steam-377840.log)

**Comment by kisak-valve** [comment]
Hello @sekoku, please copy the contents of Steam Runtime Diagnostics from Steam (`Steam` -> `Help` -> `Steam Runtime Diagnostics`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report. I suspect that the 32 bit Vulkan render path is not healthy on your system.

**Comment by Reflex13371** [comment]
I also would not like to bump this old thread however I am encountering a weird bug. I have no clue if this issue is because of how wayland interacts with specific compositors such as Niri (Which I use) through xwayland-satellite, but any proton higher than 7.0-6, whenever you try to move with a key on a keyboard the game drops in frames down to 1fps when held. Every other input for the game works flawlessly, IE controller and the mouse. I have spent days trying to solve this issue, only to then find many people suggesting to use proton 7.0-6, however it would be great if it can work on newer protons as well.

**Comment by Reflex13371** [comment]
It may not just be FF9, it could be broken on other older unity games as well, this is just the only game I have that has encountered an issue like this

**Comment by WarpedEdge** [comment]
I want to add what @Reflex13371 mentioned about as I was troubleshooting with them. Them switching between KDE/Niri or EndeavourOS or Bazzite does not matter, it always has the same issue, but its even more odd as it happens on multiple devices of theirs, yet others do not have this issue so its definitely narrowed down to something within later protons and maybe the hardware? Had them reimage to Windows and it works fine there, but back to a Linux distro is occurs all over again.

**Comment by Reflex13371** [comment]
Digging through issues, it seems awfully similar to #5294 but it didn't get fixed when swapping to KDE plasma, not sure if it's because of only getting the plasma-desktop as a backup to niri, but that also doesn't seem to be the case due to happening on other distros and DE I tested in

**Comment by Reflex13371** [comment]
I also would like to mention if holding a key with ctrl, it works completely fine, this issue is intriguing me more everything new I find that I have never experienced. Tried shift as another modifier key, and it still lagged

**Comment by Reflex13371** [comment]
> I also would like to mention if holding a key with ctrl, it works completely fine, this issue is intriguing me more everything new I find that I have never experienced. Tried shift as another modifier key, and it still lagged

I was wrong, only solved the lag on the menu, gameplay still slows whenever a key is pressed, even with ctrl

**Comment by codeweaverwill** [comment]
@Reflex13371 Hm, I haven't been able to repro so far. Could you attach a Proton log and your system information?

**Comment by Reflex13371** [comment]
Yes @codeweaverwill, I can absolutely do that, it happens on all my systems so I'll do both of the ones I mainly use being my laptop I use for work and my main PC but it will need to be a bit as I am busy for the day

**Comment by sekoku** [comment]
> Hello [@sekoku](https://github.com/sekoku), please copy the contents of Steam Runtime Diagnostics from Steam (`Steam` -> `Help` -> `Steam Runtime Diagnostics`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report. I suspect that the 32 bit Vulkan render path is not healthy on your system.

Oops, I just saw this. I can absolutely do that but I wouldn't get why it would be corrupt other than Arch ([meme response]) being an issue.

[Here is the gist log](https://gist.github.com/sekoku/ab55e71eb55402f9732dc1cd5348e48d), let me know if that helps you track down the issue, @kisak-valve . Sorry again for the delayed response.

**Comment by kisak-valve** [comment]
https://gist.github.com/sekoku/ab55e71eb55402f9732dc1cd5348e48d#file-gistfile1-txt-L1513-L1526 is the section to focus on. I'd double check that the `lib32-vulkan-radeon` system package is installed.

**Comment by Reflex13371** [comment]
Sorry for the late comment, haven't figured out the best way to provide both logs and system information in one message.

**Comment by Reflex13371** [comment]
@codeweaverwill 

> Sorry for the late comment, haven't figured out the best way to provide both logs and system information in one message.

**Comment by sekoku** [comment]
> https://gist.github.com/sekoku/ab55e71eb55402f9732dc1cd5348e48d#file-gistfile1-txt-L1513-L1526 is the section to focus on. I'd double check that the `lib32-vulkan-radeon` system package is installed.

That worked. Not sure why it isn't in the default Arch install. But good to know. I'll put this on ProtonDB for others because I know this has been a common problem for Out Of Box experience for end users.

**Comment by codeweaverwill** [comment]
> Sorry for the late comment, haven't figured out the best way to provide both logs and system information in one message.

For the system info, please copy the contents of Steam Runtime Diagnostics from Steam (`Steam` -> `Help` -> `Steam Runtime Diagnostics`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

For the logs, please add [PROTON_LOG=1 %command%](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options, reproduce the regression, and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.)

Edit: I'm just copy-pasting the always helpful Kisak directions from earlier in the thread, don't mind the initially included mis-copied suspicions :p

**Comment by Reflex13371** [comment]
Should have these for tomorrow or Sunday, I will make sure I can make time to post these as I do really appreciate that you are willing to help. It is difficult when busy though to find time

**Comment by AJuujarvi** [comment]
@Reflex13371 Hi there, I also tried to reproduce and was unable to. As some additional information that may be helpful, are there any mods you might be using? I know there are a few mods that are popular for this title like Memoria and Mugori that can affect the framerate of this title.

Could you also try as a launch option for the game `PROTON_USE_XALIA=0 %command%` please and let us know if that makes a difference?
